src/jvmMain/kotlin/io/beatmaps/api/bookmark.kt

```diff
@@ -0,0 +1,151 @@
+package io.beatmaps.api
+
+import de.nielsfalk.ktor.swagger.DefaultValue
+import de.nielsfalk.ktor.swagger.Ignore
+import de.nielsfalk.ktor.swagger.get
+import de.nielsfalk.ktor.swagger.ok
+import de.nielsfalk.ktor.swagger.post
+import de.nielsfalk.ktor.swagger.responds
+import de.nielsfalk.ktor.swagger.version.shared.Group
+import io.beatmaps.common.api.EPlaylistType
+import io.beatmaps.common.db.wrapAsExpressionNotNull
+import io.beatmaps.common.dbo.Beatmap
+import io.beatmaps.common.dbo.Playlist
+import io.beatmaps.common.dbo.PlaylistMap
+import io.beatmaps.common.dbo.User
+import io.beatmaps.common.dbo.complexToBeatmap
+import io.beatmaps.util.cdnPrefix
+import io.ktor.http.HttpStatusCode
+import io.ktor.server.application.call
+import io.ktor.server.locations.Location
+import io.ktor.server.locations.options
+import io.ktor.server.response.header
+import io.ktor.server.response.respond
+import io.ktor.server.routing.Route
+import org.jetbrains.exposed.sql.JoinType
+import org.jetbrains.exposed.sql.and
+import org.jetbrains.exposed.sql.deleteWhere
+import org.jetbrains.exposed.sql.insertIgnore
+import org.jetbrains.exposed.sql.insertIgnoreAndGetId
+import org.jetbrains.exposed.sql.select
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.jetbrains.exposed.sql.update
+
+@Location("/api")
+class BookmarksApi {
+    @Group("Bookmarks")
+    @Location("/bookmarks/add")
```

Comment:
I think both playlist map changes and followers use a single endpoint with the add/remove being part of the request.
This would be ok but it's inconsistent.

Reply:
Does the same go for the option endpoints?